content/source/docs/plugin/framework/resources.html.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: "extend"
+page_title: "Plugin Development - Framework: Resources"
+description: |-
+  How to build resources in the next-generation provider development framework.
+---
+
+# Resources
+
+[Resources](/docs/language/resources/index.html) are an abstraction that allow Terraform to manage infrastructure objects, such as a compute instance, an access policy, or disk. Terraform assumes that every resource:
+
+-  operates as a pure key/value store, with values getting returned exactly as they were written.
+- needs only one API call to update or return its state.
+- can be be created, read, updated, and deleted.
+
+Providers act as a translation layer between Terraform and an API, offering one or more resources for practitioners to define in a configuration. To create resources for your provider, you need to define both the resource archetype and actions on specific resource instances.
+
+## Define Resource Archetype
+
+Implement the [`tfsdk.ResourceType`
+interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ResourceType) for every type of
+resource you want to support: compute instances, disks, access policies, etc. It
+allows you to describe the resource archetype, which is the functionality related to all instances of that resource type in the configuration, state, plan, and API. `ResourceType` has the following methods:
+
+### GetSchema
+
+`GetSchema` returns a [schema](/docs/plugin/framework/schemas.html) describing what fields are available in the resource's configuration and state.
+
+### NewResource
+`NewResource` returns a new instance of that resource type. It instantiates a new [`tfsdk.Resource` implementation](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#Resource)
+that expects to operate on data with the structure defined in `GetSchema`. It does not need to create the instance on the API.
+
+The `NewResource` method is passed a [`tfsdk.Provider` implementation](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#Provider).
+This is the [provider type](/docs/plugin/framework/providers.html) after its
+`Configure` method was called. The `NewResource` method can type-assert on this
+and inject it into the `Resource`, allowing the `Resource` to have
+strongly-typed access to the configured API client or other provider
+configuration data.
+
+<br>
+**Example**
+
+```go
+type computeInstanceResourceType struct{}
+
+func (c computeInstanceResourceType) GetSchema(_ context.Context)
```

Comment:
I want to note that I saw in the engineering style guide that we're supposed to break long lines at 80 characters so that there's no scroll bar in the code block. So I did that here. If you do not like it, or it's wrong somehow, please let me know and I'll put it back! But wanted to flag this for you.